src/graphics/Screen.cpp

```diff
@@ -1206,7 +1206,7 @@ void Screen::setFrames()
     LOG_DEBUG("Total frame count: %d\n", totalFrameCount);
 
     // We don't show the node info our our node (if we have it yet - we should)
-    size_t numnodes = nodeStatus->getNumTotal();
+    size_t numnodes = nodeDB.getNumNodes();
```

Comment:
This is coming from the same source of truth. What is the reason you move the call one step down? This can also be made current with a forceUpdate call iirc...

Reply:
The reason for the problem is the order of initialisation in main.cpp and the interdependencies between the two modules nodeDB and screen.
nodeDB is initialized first, creates the nodes and notifies observers. At that time screen is not subscribed because it's not initialized. When the observers in screen are initialized, it takes time to update the node number until the first notifiy arrives (typically within one minute, sometimes much longer). The change in the code will immediately display the right number of known nodes (due to PR #2404).

The forceUpdate works in the other direction, from nodeDB to screen. But how could screen ask for an update via that mechanism?
As screen is already directly(!) accessing attributes of nodeDB (updateGUI), I think for the sake of simplicity and user-friendlyness that is gained screen may also request the node number directly.

Reply:
yes, i was thinking backwards here, good to go ... :-)